packages/context/test/acceptance/method-level-bindings.feature.md

```diff
@@ -0,0 +1,40 @@
+# Feature: Context bindings - injecting dependencies of methods
+
+- In order to receive information from the context for a method
+- As a developer
+- I want to setup bindings for my method
+- So that method dependencies are injected by the IoC framework
+
+## Scenario: Inject method arguments
+
+ - Given a context
+ - Given class `InfoController`
+ - Given a class binding named `controllers.info` bound to class `InfoController`
+ - When I resolve the binding for `controllers.info`
+ - Then I get a new instance of `InfoController`
+ - When I invoke the `hello` method, the parameter `user` is resolved to the
+ - value bound to `user` key in the context
+
+ ```ts
+ const ctx = new Context();
+ // Mock up user authentication
+ ctx.bind('user').toDynamicValue(() => Promise.resolve('John'));
+
+ class InfoController {
+
+   hello(@inject('user') user: string):string {
+     const msg = `Hello ${user}`;
+     console.log(msg);
+     return msg;
+   }
+ }
+
+ ctx.bind('controllers.info').toClass(InfoController);
+
+ const instance = await ctx.get('controllers.info');
+ const msg = await invokeMethod(instance, 'hello', ctx); // Hello John
+ ```
+
+## Notes
+
+- static method injection is not supported
```

Comment:
I don't mind this limitation, as long as the UX is good. What happens when the user attempts to invoke a static method? Will they receive a helpful error message? Please add a test to verify.